src/poetry/utils/authenticator.py

```diff
@@ -193,7 +195,8 @@ def authenticated_url(self, url: str) -> str:
     def request(
         self, method: str, url: str, raise_for_status: bool = True, **kwargs: Any
     ) -> requests.Response:
-        headers = kwargs.get("headers")
+        headers = kwargs.get("headers", {}) or {}
```

Comment:
`or {}` isn't doing anything useful

cf https://github.com/python-poetry/poetry/blob/02448cf7f184dea204156f7dcb620a4f01a0068e/src/poetry/publishing/uploader.py#L82, which does this on the session - it would be consistent here to do the work in `get_session()`.

Reply:
I think `or {}` covers the potential of there being a `headers` key in the `dict` with a `None` value. 


I will apply the UA to the session itself. Thank you for the suggestion.